src/generate/index.js

```diff
@@ -88,12 +87,25 @@ function parseBlocks (blocks, locations) {
     comment: '#'
   })
     .then((parsed) => {
-      var lastEnd = 0
+      let lastEnd = 0
 
       return _.reduce(parsed, (acc, row) => {
-        var start = Number(row.startIpNum)
-        var end = Number(row.endIpNum)
-        var locid = row.locId
+        const { firstAddress, lastAddress } = ip.cidrSubnet(row.network)
+        const start = ip.toLong(firstAddress)
+        const end = ip.toLong(lastAddress)
+
+        const { geoname_id } = row // eslint-disable-line camelcase
+        const geonameData = locations[geoname_id]
+
+        // conform to legacy input format by filling up missing data the first time
+        // a geoname is inspected
+        if (Array.isArray(geonameData) && geonameData.length < 7) {
+          geonameData.push(
+            String(row.postal_code),
+            Number(row.latitude),
+            Number(row.longitude)
```

Comment:
📝Last two could end up as `NaN`s if inputs aren't valid numbers. If that is not valid output it would be worth to uphold invariant here.